fiftyone/utils/hrm2.py

```
@@ -0,0 +1,2550 @@
+            defaults to fo.config.model_zoo_dir/hrm2. Pass False to disable
+            file export (in-memory labels only)
+
+    Returns:
+        the grouped dataset
+    """
+    # Reject grouped datasets upfront
+    if sample_collection.media_type == "group":
+        raise ValueError(
+            "apply_hrm2_to_dataset_as_groups() expects a non-grouped dataset. "
+            "The input dataset has media_type='group'."
+        )
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)